packages/kit/src/runtime/client/renderer.js

```diff
@@ -275,8 +275,15 @@ export class Renderer {
 			this._init(navigation_result);
 		}
 
-		if (!opts?.keepfocus) {
-			document.body.focus();
+		if (opts) {
+			if (!opts.keepfocus) {
+				document.body.focus();
+			}
+
+			// Scroll to top so we can compare the pageYOffset below. We cannot
+			// compare by recording the pageYOffset here as there is a possiblity
+			// it will change, e.g. different scroll heights
```

Comment:
what do you mean by different scroll heights? do you mean different page heights?

could you use the minimum of `pageYOffset` and the page height?

Reply:
Yeah page heights sounds clearer :+1: 

> could you use the minimum of pageYOffset and the page height?

I'm not too sure what you mean here, are you talking about the words to use for documentation?

Reply:
Fixed in 47ea554